packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -901,11 +903,13 @@ function applyAttributes(ctx: ParserCtx, parsedAttr: ParsedAttribute, element: B
             element.attributes.push(attr);
         } else {
             const propName = attributeToPropertyName(name);
-            element.properties.push(ast.property(propName, attr.name, attr.value, attr.location));
+            properties.set(propName, ast.property(propName, name, attr.value, attr.location));
```

Comment:
It would be great if the compiler reports a warning whenever it detects different attributes mapping to the same prop.